fix age computations on GC and a bug in C++

tbg opened the pull request:
updated `updateStatsOnGC` to correctly use `AgeTo`.
Updated `TestMVCCStatsWithRandomRuns` to not use full second timestamps.
This is the first test which does so, and it gives new (semantic) coverage
which caught a bug in the C++ code which led to incorrect age increments.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3905)

<!-- Reviewable:end -->

RaduBerinde commented:
*[storage/engine/rocksdb/db.cc, line 1490 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3905#-K8AzNc4LqCt15K4ZnJB:-K8AzNc4LqCt15K4ZnJC:-661997170) ([raw file](https://github.com/cockroachdb/cockroach/blob/be0a2f0c7d48e0002fa5ad0347ff5df6535386f4/storage/engine/rocksdb/db.cc#L1490)):*
1e9 is a double which made the entire computation be in floating point and then truncated to an integer. 3749632704/1e9 - 2787942144/1e9 is ~0.9

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3905)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
*[storage/engine/rocksdb/db.cc, line 1490 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3905#-K8AzNc4LqCt15K4ZnJB:-K8B-17okJ3LQSoh2MOt:-1555829008) ([raw file](https://github.com/cockroachdb/cockroach/blob/be0a2f0c7d48e0002fa5ad0347ff5df6535386f4/storage/engine/rocksdb/db.cc#L1490)):*
BTW it makes code easier to read if we define constants like NS_PER_SEC and use those. That makes it evident what the division is doing without any comments.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3905)*
<!-- Sent from Reviewable.io -->

petermattis commented:
LGTM

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[storage/engine/rocksdb/db.cc, line 1490 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3905#-K8AzNc4LqCt15K4ZnJB:-K8C96mzVhRRDksNA34Z:-1005443665) ([raw file](https://github.com/cockroachdb/cockroach/blob/be0a2f0c7d48e0002fa5ad0347ff5df6535386f4/storage/engine/rocksdb/db.cc#L1490)):*
What @radu said. Our C++ style would give the constant a name like `kNanosecondPerSecond` or something like that. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3905#-:-K8C9jepA765WurfScvw:676834233)*
<!-- Sent from Reviewable.io -->

tbg committed:
fix age computations on GC and a bug in C++

updated `updateStatsOnGC` to correctly use `AgeTo`.
Updated `TestMVCCStatsWithRandomRuns` to not use full second timestamps.
This is the first test which does so, and it gives new (semantic) coverage
which caught a bug in the C++ code which led to incorrect age increments.

tschottdorf commented:
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[storage/engine/rocksdb/db.cc, line 1490 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3905#-K8AzNc4LqCt15K4ZnJB:-K8Pzzv1L_1iIEEBov7Y:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/be0a2f0c7d48e0002fa5ad0347ff5df6535386f4/storage/engine/rocksdb/db.cc#L1490)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3905)*
<!-- Sent from Reviewable.io -->
